File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/VisualBasic/Silverlight/2052/BusinessApplication/Models/User.partial.vb
Imports System
Imports System.ComponentModel

Namespace Web

    Partial Public Class User
#Region "使 DisplayName 可绑定"

        ''' <summary>
        ''' 用于在 <see cref="User.DisplayName"/> 更改时生成属性更改通知的 <c>OnPropertyChanged</c> 方法重写。
        ''' </summary>
        ''' <param name="e">属性更改事件参数。</param>
        Protected Overrides Sub OnPropertyChanged(ByVal e As System.ComponentModel.PropertyChangedEventArgs)
            MyBase.OnPropertyChanged(e)

            If e.PropertyName = "Name" Or e.PropertyName = "FriendlyName" Then
                Me.RaisePropertyChanged("DisplayName")
            End If
        End Sub
#End Region
    End Class
End Namespace

File: src/ChpokkWeb/SystemFiles/Templates/ProjectTemplates/CSharp/Silverlight/1041/BusinessApplication/Models/User.partial.cs
namespace $safeprojectname$.Web
{
    using System;
    using System.ComponentModel;

    /// <summary>
    /// <see cref="User"/> クラスの拡張です。
    /// </summary>
    public partial class User
    {
        #region DisplayName をバインド可能にします

        /// <summary>
        /// <see cref="User.DisplayName"/> が変更された場合にプロパティの変更通知を生成する <c>OnPropertyChanged</c> メソッドのオーバーライドです。
        /// </summary>
        /// <param name="e">プロパティ変更イベントの引数です。</param>
        protected override void OnPropertyChanged(PropertyChangedEventArgs e)
        {
            base.OnPropertyChanged(e);

            if (e.PropertyName == "Name" || e.PropertyName == "FriendlyName")
            {
                this.RaisePropertyChanged("DisplayName");
            }
        }
        #endregion
    }
}
